.vscode/settings.json

```diff
@@ -1,6 +1,46 @@
 {
+  "editor.formatOnSave": true,
+  "editor.defaultFormatter": "esbenp.prettier-vscode",
+  "eslint.validate": [
+    "javascript",
+    "javascriptreact",
+    "typescript",
+    "typescriptreact",
+    "vue"
+  ],
+  "[javascript]": {
+    "editor.codeActionsOnSave": {
+      "source.fixAll.eslint": true
+    },
     "editor.formatOnSave": false,
-    "editor.defaultFormatter": "esbenp.prettier-vscode",
-    "prettier.jsxSingleQuote": true,
-    "prettier.singleQuote": true
+  },
+  "[javascriptreact]": {
+    "editor.codeActionsOnSave": {
+      "source.fixAll.eslint": true
+    },
+    "editor.formatOnSave": false
```

Comment:
Do we need this additional `editor.formatOnSave: false`?

Reply:
The problem for me is that the prettier extension's format on save does not result in the same format as the eslint --fix. First, I thought this was a problem with the ESLint config. However, using the prettier cli, i.e. `prettier --write some/file.ts`, matches the output from ESLint. This means the Prettier extension does not behave the same as the cli. So far, I have not been able to figure out how to fix this. This is surprising as the the prettier extension should use the local prettier version and config. For some reason, this does not work as expected.

Reply:
Nevermind, the reason was that the default formatter was overridden for some languages in my user vscode config. This then trumps the general default formatter setting